apps/oozie/src/oozie/models2.py

```diff
@@ -3775,11 +3837,17 @@ def _save_workflow(workflow, layout, user, fs=None):
   if workflow.get('id'):
     workflow_doc = Document2.objects.get(id=workflow['id'])
   else:
-    workflow_doc = Document2.objects.create(name=workflow['name'], uuid=workflow['uuid'], type='oozie-workflow2', owner=user, description=workflow['properties']['description'])
-    Document.objects.link(workflow_doc, owner=workflow_doc.owner, name=workflow_doc.name, description=workflow_doc.description, extra='workflow2')
+    workflow_doc = Document2.objects.create(
+      name=workflow['name'], uuid=workflow['uuid'], type='oozie-workflow2', owner=user, description=workflow['properties']['description']
+    )
+    Document.objects.link(
+      workflow_doc, owner=workflow_doc.owner, name=workflow_doc.name, description=workflow_doc.description, extra='workflow2'
+    )
 
   # Excludes all the sub-workflow and Hive dependencies. Contains list of history and coordinator dependencies.
-  workflow_doc.dependencies = workflow_doc.dependencies.exclude(Q(is_history=False) & Q(type__in=['oozie-workflow2', 'query-hive', 'query-java']))
+  workflow_doc.dependencies.set(
```

Comment:
Adding set here also. Please review it again (other than this all are pylint fixes). @romainr

Reply:
:+1: